File: springmvc/spring-springmvc-hibernate/src/main/java/com/jiangKlijna/web/dao/IDao.java
package com.jiangKlijna.web.dao;

import java.io.Serializable;
import java.util.List;

public interface IDao {
    void close();

    org.hibernate.Query query(String hql);

    org.hibernate.Criteria criteria(Class<?> arg0);

    org.hibernate.SQLQuery sqlQuery(String sql);

    void save(Object obj);

    void update(Object obj);

    void delete(Object obj);

    Object get(Class<?> entityClazz, Serializable id);

    long findCount(Class<?> entityClazz);

    List<?> find(String hql);

    List<?> find(String hql, Object... params);

    List<?> findByPage(String hql, int pageNo, int pageSize);

    List<?> findByPage(String hql, int pageNo, int pageSize, Object... params);
}

File: spring-springmvc-hibernate-kotlin/src/main/java/com/jiangKlijna/web/dao/IDao.kt
package com.jiangKlijna.web.dao

import java.io.Serializable

interface IDao {
    fun close()

    fun query(hql: String): org.hibernate.Query<*>

    fun criteria(arg0: Class<*>): org.hibernate.Criteria

    fun sqlQuery(sql: String): org.hibernate.SQLQuery<*>

    fun save(obj: Any)

    fun update(obj: Any)

    fun delete(obj: Any)

    operator fun <T> get(entityClazz: Class<*>, id: Serializable): T

    fun findCount(entityClazz: Class<*>): Long

    fun find(hql: String): List<*>

    fun find(hql: String, vararg params: Any): List<*>

    fun findByPage(hql: String, pageNo: Int, pageSize: Int): List<*>

    fun findByPage(hql: String, pageNo: Int, pageSize: Int, vararg params: Any): List<*>
}
